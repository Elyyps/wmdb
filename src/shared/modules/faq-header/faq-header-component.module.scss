.faq-header {
  position: relative;
  overflow: hidden;
  background: linear-gradient(45deg, #34aadf 0%, #02906f 100%);
  color: $global-inverse-color;
  @media (min-width: $breakpoint-large) {
    font-size: 20px;
  }
  @media (max-width: $breakpoint-small-max) {
    text-align: center;
  }
  @media (max-width: $breakpoint-xsmall-max) {
    font-size: 14px;
  }

  &__holder {
    @media (min-width: $breakpoint-medium) {
      display: flex;
      align-items: center;
    }
  }
  &__image {
    @media (min-width: $breakpoint-medium) {
      width: 30%;
      flex: 0 0 33%;
      height: 252px;
      display: flex;
      align-items: flex-end;
      justify-content: center;
      order: -1;
    }
    @media (max-width: $breakpoint-small-max) {
      max-width: 200px;
      height: 100px;
      margin-top: 20px;
      img {
        width: 100%;
        height: 100%;
        object-fit: fill;
      }
    }
  }
  &__body {
    padding: 32px 0;
    flex: 1 1 auto;
    @media (min-width: $breakpoint-medium) {
      padding-right: 25%;
    }
    @media (max-width: $breakpoint-small-max) {
      padding-bottom: 0;
    }
    h1 {
      color: $global-inverse-color;
    }
  }
}
