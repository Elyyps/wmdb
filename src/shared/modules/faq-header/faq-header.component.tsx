import * as React from "react";
import styles from "./faq-header-component.module.scss";

export interface IFAQHeaderComponentProps {
  content?: string;
  image?: string;
  title: string;
}

const FAQHeaderComponent = (props: IFAQHeaderComponentProps) => (
  <div className={styles["faq-header"]}>
    <div className="uk-container">
      <div className={styles["faq-header__holder"]}>
        <div className={styles["faq-header__body"]}>
          <h1>{props.title}</h1>
          {props.content}
        </div>
        {props.image && (
          <div className={styles["faq-header__image"]}>
            <img src={props.image} alt="image" />
          </div>
        )}
      </div>
    </div>
  </div>
);

export { FAQHeaderComponent };
