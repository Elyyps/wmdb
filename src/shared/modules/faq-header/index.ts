export * from "./faq-header.component";
