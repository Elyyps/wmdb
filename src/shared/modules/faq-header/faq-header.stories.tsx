import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FAQHeaderComponent } from "./faq-header.component";
import PlaceholderImage2 from "@assets/img03.png";
storiesOf("FAQHeader", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <FAQHeaderComponent
      title={"Veelgestelde vragen"}
      content={"Alle vragen omtrent WegmetdeBaas. Hoe vind je zo snel mogelijk het perfecte uitje?"}
      image={PlaceholderImage2}
    />
  ));
