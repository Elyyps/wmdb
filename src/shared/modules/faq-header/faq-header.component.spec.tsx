import React from "react";
import { shallow } from "enzyme";
import { FAQHeaderComponent } from "./faq-header.component";
import PlaceholderImage2 from "@assets/img03.png";

describe("[FAQHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <FAQHeaderComponent
        title={"Veelgestelde vragen"}
        content={"Alle vragen omtrent WegmetdeBaas. Hoe vind je zo snel mogelijk het perfecte uitje?"}
        image={PlaceholderImage2}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
