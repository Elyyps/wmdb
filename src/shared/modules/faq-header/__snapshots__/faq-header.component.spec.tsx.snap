// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FAQHeader] should render component without crashing 1`] = `
<div
  className="faq-header"
>
  <div
    className="uk-container"
  >
    <div
      className="faq-header__holder"
    >
      <div
        className="faq-header__body"
      >
        <h1>
          Veelgestelde vragen
        </h1>
        Alle vragen omtrent WegmetdeBaas. Hoe vind je zo snel mogelijk het perfecte uitje?
      </div>
      <div
        className="faq-header__image"
      >
        <img
          alt="image"
          src="img03.png"
        />
      </div>
    </div>
  </div>
</div>
`;
